### PR TITLE
Fixes gunboots shooting the wearer

### LINKED ISSUE #58242: Bullets fired from gunboots hit the wearer.
## Round ID:
[159826](https://scrubby.melonmesa.com/round/159826)

## Reproduction:
Attempt to walk around with the gunboots equipped for more than 5 seconds.

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
For whatever reason, gunboots could immediately shoot the person wearing them when they fired. This obviously isn't ideal, so this fixes that. Once the bullet has ricocheted off something, of course, they're once again fair game to be hit. This also kills the projectile_shooter component because it was pointless
Fixes: #58242

## Why It's Good For The Game
As if the trail of people trying to kill you for wearing this isn't bad enough, you don't need the shoes themselves doing you in

## Changelog
:cl: Ryll/Shaps
fix: You can no longer shoot yourself in the foot with gunboots
/:cl:

### CODE REVIEW

**Comment by dondonpow** [comment]
Holy shit thank you.

File: code/datums/components/projectile_shooter.dm [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
 	src.projectile_type = projectile_type
 	src.shot_prob = shot_prob
+	if(istype(restricted_zones))
+		src.restricted_zones = restricted_zones
 
 	// unarmed attack is a special case for gunboots so they can shoot someone you stomp on. This happens 100% of the time you kick, ignoring shot_prob
 	if(islist(signal_or_sig_list) && (COMSIG_HUMAN_MELEE_UNARMED_ATTACK in signal_or_sig_list))
 		signal_or_sig_list -= COMSIG_HUMAN_MELEE_UNARMED_ATTACK
-		RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, .proc/check_equip)
-		RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/check_dropped)
```
Comment (LemonInTheDark) [inline code/datums/components/projectile_shooter.dm:24]: I take it this was the only thing that used this component?

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 2]
```diff
@@ -540,7 +540,21 @@
 	var/projectile_type = /obj/projectile/bullet/c10mm
 	/// Each step, this is the chance we fire a shot
 	var/shot_prob = 50
+	/// Storing the projectile shooter component so we can create/remove it on equip/drop
+	var/datum/component/projectile_shooter/shooter_component
```
Comment (LemonInTheDark) [inline code/modules/clothing/shoes/miscellaneous.dm:544]: shouldn't the component handle this? why can't it?

**Comment by Ryll-Ryll** [comment]
@LemonInTheDark honestly I'm not even sure why I made the projectile_shooter component, looking back over it while working on this made me realize how sucky it is. I'm probably just going to axe it and just put the bullet shooting on gunboots stepping themselves

**Comment by AnturK** [comment]
So this done, or are you working on axing this.

**Pull request merged**